design-system/src/theme-provider.tsx

```diff
@@ -92,19 +92,47 @@ ThemeProvider.defaultProps = {
   theme: 'default',
 };
 
-const useTheme = (parentSelector = defaultParentSelector): ThemeName => {
+type TUseThemeResult = {
+  currentTheme: ThemeName;
+};
+const useTheme = (parentSelector = defaultParentSelector): TUseThemeResult => {
   const [theme, setTheme] = useState<ThemeName>('default');
   const parentSelectorRef = useRef(parentSelector);
+  const observerRef = useRef(
+    new MutationObserver((mutationList, _observer) => {
+      // Since we are only observing the theme DOM node for changes in its
+      // `data-theme` attribute (configured below when calling `observe` function),
+      // we will receive a single element in the list
+      setTheme(
+        (mutationList[0].target as HTMLElement).dataset.theme as ThemeName
+      );
```

Comment:
Wondering if we should access these properties in a more safer way.

For example, looking at the [MutationObserver docs](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver/MutationObserver#the_callback_function) they iterate on the `mutationList` and check the mutation `type`.

What do you think?

Reply:
We can do that but I thought is not necessary since we are observing only a change on the `data-theme` attribute, we are not supposed to receive another mutations.

Reply:
I guess it's fine, just wanted to double check and be more on the safe side.